data/modules/CargoRun/CargoRun.lua

```diff
@@ -0,0 +1,1148 @@
+			cargoname	= l[custom_cargo[ad.branch][ad.cargotype].l10n_key],
+			starport	= ad.location:GetSystemBody().name,
+			system		= ad.location:GetStarSystem().name,
+			sectorx		= ad.location.sectorX,
+			sectory		= ad.location.sectorY,
+			sectorz		= ad.location.sectorZ,
+			dom_starport	= ad.domicile:GetSystemBody().name,
+			dom_system	= ad.domicile:GetStarSystem().name,
+			dom_sectorx	= ad.domicile.sectorX,
+			dom_sectory	= ad.domicile.sectorY,
+			dom_sectorz	= ad.domicile.sectorZ,
+			dist	  = string.format("%.2f", ad.dist),
+		})
+		form:SetMessage(introtext)
+
+	elseif option == 1 then
+		local whysomuchtext = l["WHYSOMUCHTEXT__" .. ad.branch .. "__" .. ad.cargotype] or l["WHYSOMUCHTEXT__" .. ad.branch]
+
+		if whysomuchtext then
+			form:SetMessage(whysomuchtext)
+		elseif ad.urgency >= 0.8 then
+			form:SetMessage(l["WHYSOMUCHTEXT_URGENT_" .. Engine.rand:Integer(1, getNumberOf("WHYSOMUCHTEXT_URGENT"))])
+		else
+			form:SetMessage(l["WHYSOMUCHTEXT_" .. Engine.rand:Integer(1, getNumberOf("WHYSOMUCHTEXT"))])
+		end
+
+	elseif option == 2 then
+		local howmuch
+		if ad.wholesaler then
+			howmuch = string.interp(l["HOWMUCH_WHOLESALER_" .. Engine.rand:Integer(1, getNumberOf("HOWMUCH_WHOLESALER"))], {
+				cargo	  = ad.cargo,
+				cargoname = l[custom_cargo[ad.branch][ad.cargotype].l10n_key],
+			})
+		else
+			num = ad.cargo <= 3 and getNumberOf("HOWMUCH") or (getNumberOf("HOWMUCH") - 2) -- The last 2 strings contain "only"
+			howmuch = string.interp(l["HOWMUCH_" .. Engine.rand:Integer(1, num)], {
+				cargo	  = ad.cargo,
+			})
+		end
+		form:SetMessage(howmuch)
+
+	elseif option == 3 then
+		if Game.player.freeCapacity < ad.cargo then
+			form:SetMessage(l.YOU_DO_NOT_HAVE_ENOUGH_CARGO_SPACE_ON_YOUR_SHIP)
+			return
+		end
+		if not ad.pickup then
+			Game.player:AddEquip(custom_cargo[ad.branch][ad.cargotype], ad.cargo, "cargo")
+			cargo_picked_up = true
+		else
+			cargo_picked_up = false
+		end
+		form:RemoveAdvertOnClose()
+		ads[ref] = nil
+		local mission = {
+			type		= "CargoRun",
+			domicile	= ad.domicile,
+			client		= ad.client,
+			location	= ad.location,
+			localdelivery	= ad.localdelivery,
+			wholesaler	= ad.wholesaler,
+			pickup		= ad.pickup,
+			cargo_picked_up	= cargo_picked_up,
+			introtext	= ad.introtext,
+			risk		= ad.risk,
+			reward		= ad.reward,
+			due		= ad.due,
+			cargo		= ad.cargo,
+			branch		= ad.branch,
+			cargotype	= ad.cargotype,
+		}
+		table.insert(missions,Mission.New(mission))
+		if ad.pickup then
+			form:SetMessage(l["ACCEPTED_PICKUP_" .. Engine.rand:Integer(1, getNumberOf("ACCEPTED_PICKUP"))])
+		else
+			form:SetMessage(l["ACCEPTED_" .. Engine.rand:Integer(1, getNumberOf("ACCEPTED"))])
+		end
+		return
+
+	elseif option == 4 then
+		local num = math.floor(ad.urgency * (getNumberOf("URGENCY") - 1)) + 1
+		local urgency = l["URGENCY__" .. ad.branch .. "__" .. ad.cargotype] or l["URGENCY__" .. ad.branch] or l["URGENCY_" .. num]
+		form:SetMessage(urgency .. Format.Date(ad.due))
```

Comment:
If possible, I think you should use the `{date}` construct so that other languages can put the time string anywhere in the string